### DISCUSSION TITLE
Arm32v7 samples do not build.

### DISCUSSION THREAD

**Discussion by opcfoundation-org** [dotnet/dotnet-docker#3963] [Q&A] [answered]
Replication:

git clone https://github.com/dotnet/dotnet-docker.git
cd dotnet-docker\samples\dotnetapp
docker build --platform linux/arm/v7 -t dotnetapp .

Error:

```
=> ERROR [build 4/6] RUN dotnet restore                                                                           0.5s
 => [stage-1 2/3] WORKDIR /app                                                                                     0.0s
------
 > [build 4/6] RUN dotnet restore:
#11 0.466 A fatal error occurred, the folder [/usr/share/dotnet/host/fxr] does not contain any version-numbered child folders
------
```

I ran into this when my own working images that I was trying to get running on a RaspberryPI.
 
All of the online help for that error says my dotnet installation is broken but I am using the MS supplied docker so something is really messed up.

**Comment by mthalman** [comment, marked as answer]
I'm assuming you're attempting to build this on an x64 machine? That uses QEMU to run the code in emulation which has incompatibility with .NET. See the comment here: https://github.com/dotnet/dotnet-docker/issues/1537#issuecomment-615269150.

Instead, you can use the `Dockerfile.debian-arm32` file at that location and don't use the `--platform` option in your `docker build` command. That will allow you to build an Arm32 app from an AMD64 SDK container.

The point is that if you're doing a cross-architecture scenario like this, you need to be explicit in the tag names that you use such that retrieve the necessary architecture. The SDK tag should match the architecture of your host machine and the runtime/aspnet tag should match the architecture of your desired runtime environment.

**Reply by opcfoundation-org** [reply]
Thank you for the help.

Am I correct to infer that installing the dotnetsdk on an arm32 docker image is not supported? (the docker files seem to install the dotnetsdk on an amd64 image, cross compile and then install the outputs and runtime on an arm32 image).

**Reply by richlander** [reply]
It is supported. What environment are running `docker build` on?

**Reply by mthalman** [reply]
Yes, it's supported. If you were to build that Dockerfile on your Raspberry Pi, for example, it would work. The problem that comes is with the cross-architecture scenario as I described. That requires more explicit tag and RID usage as we provide in the architecture-specific sample Dockerfiles.
